vllm/model_executor/models/utils.py

```diff
@@ -1,22 +1,70 @@
-from typing import Dict, List, Protocol, Tuple
+from typing import Dict, Iterable, List, Optional, Protocol, Tuple
 
 import torch
+import torch.nn as nn
 from torch.func import functional_call
+from transformers import PretrainedConfig
 
+from vllm.config import (CacheConfig, LoRAConfig, MultiModalConfig,
+                         SchedulerConfig)
+from vllm.model_executor.layers.quantization import QuantizationConfig
+from vllm.model_executor.model_loader.loader import build_model
+from vllm.model_executor.models import ModelRegistry
 from vllm.multimodal import BatchedTensors
 from vllm.utils import is_pin_memory_available
 
 
+def filter_weights(weights: Iterable[Tuple[str, torch.Tensor]], prefix: str):
+    """
+    Helper function to load weights for inner models.
+
+    See also:
+        :ref:`init_inner_model`
+    """
+    for name, loaded_weight in weights:
+        name = name.split(".")
+        if prefix == name.pop(0):
+            name = ".".join(name)
+            yield name, loaded_weight
+
+
+def init_inner_model(
```

Comment:
Why don't we name this `init_language_model`? I think the word `inner` is pretty confusing for developers. Is there a case where we use this function to initialize a non-LM model?

Reply:
This can be used for any model registered in vLLM (not just LMs), since it shares the same loading logic. The recipe introduced by #7067 can be applied to any inner model that implements weight loading logic.

Reply:
I think it's a just a bit confusing when we have a `_init_vision_tower` and `init_inner_model` together, and perhaps we can think of a better name for this pattern other than `inner`. How about explicitly calling it out as `init_registered_model`?

Reply:
Hmm... I'm thinking of `init_vllm_model`. Would that work?

Reply:
Let's do `init_vllm_registered_model` - I know it's a bit wordy, but I think the purpose of this function will be a lot easier to tell from its name.